### PR TITLE
Refactoring for symfony 5 and LTS

### PR DESCRIPTION
**Pull request by ptondereau**
- Refactor files structure
- New tests with Symfony Flex
- PHP 7 only
- Symfony LTS support

### CODE REVIEW

**Comment by ptondereau** [comment]
@florianajir should be good. Next steps are to add some good CodeStyle rules and integration with Coverall

**Comment by florianajir** [comment]
> @florianajir should be good. Next steps are to add some good CodeStyle rules and integration with Coverall

Thank you for the contribution, it looks good to me, nice work 👍 

Just wondering about the new directory structure `src` in bundle context but it seems your update about psr-4 part in `composer.json` does the work.

About code-styling we could check psr-2 as part of the PR process (in checklist). I use style.ci on other projects but I'll look into Coverall.

File: .gitignore [hunk 1]
```diff
@@ -5,10 +5,13 @@
...
+src/build/
 
 # Composer PHAR
 /composer.phar
 
 # IDE prefs
 /.idea/
+
+.phpunit.result.cache
+composer.lock
```
Comment (florianajir) [inline .gitignore]: IMO the `composer.lock` should not be ignored by versioning system, it creates a state of the dependencies versions.

File: phpunit.xml.dist [hunk 2]
```diff
@@ -1,23 +1,21 @@
...
 <phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-         xsi:noNamespaceSchemaLocation="http://schema.phpunit.de/4.8/phpunit.xsd"
+         xsi:noNamespaceSchemaLocation="http://schema.phpunit.de/7.0/phpunit.xsd"
          backupGlobals="false"
          colors="true"
          bootstrap="vendor/autoload.php">
   <testsuites>
     <testsuite name="Datanova-bundle test suite">
-      <directory>./Tests</directory>
+      <directory>./test</directory>
```
Comment (florianajir) [inline phpunit.xml.dist:10]: not sure about the directory structure when I read the official documentation recommendations https://symfony.com/doc/current/bundles/best_practices.html#directory-structure

**Comment by ptondereau** [comment]
Sorry. I mean Coverall for code coverage. But yes StyleCI is a good choice for CS

**Comment by ptondereau** [comment]
New directory structure is driven by new convention introduced by Symfony 5 and PSR guidance.

**Review by florianajir — APPROVED** [review]
ok good to know ! let's rock !

**Comment by ptondereau** [comment]
any plan for a merge and a release?

**Comment by florianajir** [comment]
> any plan for a merge and a release?

https://github.com/florianajir/datanova-bundle/releases/tag/0.4.0

**Pull request merged**